### PR TITLE
TM-1539 - Fix/cdo mapping

### PR DESCRIPTION
**Pull request by rtirserio**
Requires https://github.com/MetaPhase-Consulting/mock-fsbid/pull/40

### CODE REVIEW

File: talentmap_api/fsbid/services/client.py [hunk 1]
```diff
@@ -141,28 +141,32 @@ def get_client_csv(query, jwt_token, rl_cd, host=None):
...
+        "id": employee.get("per_seq_num", None),
+        "name": employee.get("per_full_name", None),
         "perdet_seq_number": data.get("perdet_seq_num", None),
-        "grade": data.get("grade_code", None),
-        "skills": map_skill_codes(data),
-        "employee_id": data.get("emplid", None),
-        "role_code": data.get("role_code", None),
-        "pos_location_code": data.get("pos_location_code", None),
+        "grade": employee.get("per_grade_code", None),
+        "skills": map_skill_codes(employee),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py]: I'm getting null descriptions in the skills array for clients:

```json
"skills":[{"code":"0020","description":null}]
```

File: talentmap_api/fsbid/services/client.py [hunk 2]
```diff
@@ -141,28 +141,32 @@ def get_client_csv(query, jwt_token, rl_cd, host=None):
 
 
 def fsbid_clients_to_talentmap_clients(data):
+    employee = data.get('employee', None)
+    position = data.get('currentPosition', None)
     return {
-        "id": data.get("perdet_seq_num", None),
-        "name": data.get("per_full_name", None),
+        "id": employee.get("per_seq_num", None),
+        "name": employee.get("per_full_name", None),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py]: Is `fsbid_clients_to_talentmap_clients` being in conjuction with /Clients or /Persons? /Clients doesn't give us full_name, only first, last, and middle. Whereas /Persons gives us first, last, middle and full_name.

File: talentmap_api/fsbid/services/client.py [hunk 3]
```diff
@@ -141,28 +141,32 @@ def get_client_csv(query, jwt_token, rl_cd, host=None):
...
-        "grade": data.get("grade_code", None),
-        "skills": '\n'.join(map_skill_codes_for_csv(data)),
-        "employee_id": data.get("emplid", None),
-        "role_code": data.get("role_code", None),
-        "pos_location_code": data.get("pos_location_code", None),
+        "grade": employee.get("per_grade_code", None),
+        "skills": '\n'.join(map_skill_codes_for_csv(employee)),
+        "employee_id": employee.get("per_seq_num", None),
+        "role_code": data.get("rl_cd", None),
+        "pos_location_code": position.get("pos_location_code", None),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py]: Getting `pos_loc_code: ""` for all clients

Comment (mjoyce91) [reply talentmap_api/fsbid/services/client.py]: (For both, not just the CSV function specifically)

File: talentmap_api/fsbid/services/client.py (hunk 2 above)
Comment (rtirserio) [reply talentmap_api/fsbid/services/client.py]: this should only be used with clients, ill make a new one for persons

File: talentmap_api/fsbid/services/client.py [hunk 4]
```diff
@@ -141,28 +141,34 @@ def get_client_csv(query, jwt_token, rl_cd, host=None):
...
+         "id": employee.get("pert_external_id", None),
+        "name": f"{employee.get('per_first_name', None)} {employee.get('per_last_name', None)}",
         "perdet_seq_number": data.get("perdet_seq_num", None),
-        "grade": data.get("grade_code", None),
-        "skills": '\n'.join(map_skill_codes_for_csv(data)),
-        "employee_id": data.get("emplid", None),
-        "role_code": data.get("role_code", None),
-        "pos_location_code": data.get("pos_location_code", None),
+        "grade": employee.get("per_grade_code", None),
+        "skills": '\n'.join(map_skill_codes_for_csv(employee)),
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/client.py]: In the CSV, some skills are coming through as numbers and some as strings. Also only the first skill code is coming through:

Screen Shot 2020-02-04 at 3 35 25 PM

**Comment by rtirserio** [comment]
@mjoyce91 i believe these have all been addressed with the newest version of the mock

File: talentmap_api/fsbid/services/client.py (hunk 4 above)
Comment (rtirserio) [reply talentmap_api/fsbid/services/client.py]: i updated it to use the description not the code for skills. the sheet looks like it only is showing one but they are both there. they are separated with a \n so if you check the cell they are there (or open in a text editor and not excel)

Comment (mjoyce91) [reply talentmap_api/fsbid/services/client.py]: Ah okay I see. I think that will ultimately be confusing to users, so can we change it to simply be comma separated?

Comment (rtirserio) [reply talentmap_api/fsbid/services/client.py]: @mjoyce91 - updated

**Pull request merged**